component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -296,8 +296,15 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         if (userToken.equals(contextToken)) {
             context.setProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN, "");
             context.setProperty(EmailOTPAuthenticatorConstants.EMAILOTP_ACCESS_TOKEN, "");
-            String emailFromProfile = context.getProperty(EmailOTPAuthenticatorConstants.RECEIVER_EMAIL).toString();
-            context.setSubject(AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(emailFromProfile));
+            Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+            AuthenticatedUser authenticatedUser = null;
+            for (StepConfig stepConfig : stepConfigMap.values()) {
+                authenticatedUser = stepConfig.getAuthenticatedUser();
+                if (authenticatedUser != null && stepConfig.isSubjectAttributeStep()) {
+                    break;
```

Comment:
If authenticatedUser is null don't we need to throw an exception here?

Reply:
Fixed: 2b0982e79387f70dcd498d35eb7304b42ee4ee18